### ISSUE TITLE
Tracking only bound to A tags / Massive page rendering slowdown with many links

### ISSUE THREAD

**Issue by yamlfd** [googleanalytics/autotrack#63]
I have reviewed your code because of https://www.drupal.org/node/2671716 and found some issues with the code.

You also need to bind to AREA.

```
this.delegate = delegate(document, 'a',
      'click', this.handleLinkClicks.bind(this));
```

Aside binding to all A tags like today will cause:

- [Massive page rendering slowdown with many links](https://www.drupal.org/node/336924)

**Comment by yamlfd** [comment]
SVG links also need to be tracked.

**Comment by philipwalton** [comment]
In version 1.0, users will be able to customize the selector used to determine what a "link" is. For example, in 1.0, you could do something like this:

```js
ga('require', 'outboundLinkTracker', {
  linkSelector: 'a[href], area[href]'
});
```

You can see the relevant changes in c8371cd5289a014dc6cddc93e1d7ac6af0399265. Also take a look at the [1.0-dev](https://github.com/googleanalytics/autotrack/tree/1.0-dev) branch.

I'd be open to changing what the default link selector is (currently it's just `'a'`), but at least with a config option to change it, you should be able to track whatever you want.

**Comment by philipwalton** [comment]
The problem of "massive page rendering slowdown with many links" shouldn't apply here since only a single event handler is ever added to the DOM.

**Comment by yamlfd** [comment]
Oh, great. Thanks for explaination. Wasn't aware of delegate. Not sure if this works down to IE6 what may explain why I never used it in past. Whatever, than the massive slowdown will not happen with the single document binding.

The link selector config looks useful. Thanks for these example. Just wondering that the addon has no complete list as **default**.

**Comment by philipwalton** [comment]
Autotrack currently only supports IE9 and above. It won't error in older browsers, but some features might not work (and outbound link tracking is one of them).

This is usually OK though since outbound link tracking is not a mission critical feature. And analytics (in general) are not about being 100% accurate, they're about recognizing and identifying trends and opportunities.

> The link selector config looks useful. Thanks for these example. Just wondering that the addon has no complete list as **default**.

I'm not 100% what you mean by this. The default is currently `'a'`, but users can change it to whatever they want via the config option.

**Comment by philipwalton** [comment]
Now that version 1.0 is out, this issue has been solved.

**Comment by yamlfd** [comment]
Why are you not tracking "a" and "area" as default?

**Comment by philipwalton** [comment]
I'd be open to changing the default if user want that. Can you open a new issue that's a feature request for that change, and we'll see what the feedback is?

FWIW, I don't think many users are using <area> and svg links, so I'm not sure how important it is to set that as a default for everyone. But like I said, I'll consider it if users want it.

**Comment by yamlfd** [comment]
Well svg links arenot yet seen very often, but they could exist. Is it a performance penalty to add everything that could theoretically exist?

**Comment by philipwalton** [comment]
I created #105 to track this.

**Comment by philipwalton** [comment]
Fixed via #106.

**Comment by yamlfd** [comment]
Great, thanks!
